internal/cache/filesystem.go

```diff
@@ -1,53 +1,92 @@
 package cache
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
 	"os"
 	"path"
 	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	cacheSubdir = "parrot"
 )
 
-type FileCache struct {
+type FilesystemCache struct {
 	dir string
 	ttl time.Duration
 }
 
-func NewFileCache(ttl time.Duration) (*FileCache, error) {
+func NewFilesystemCache(ttl time.Duration) (*FilesystemCache, error) {
 	dir, err := os.UserCacheDir()
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "Failed to determin OS user cache directory")
 	}
-	cacheDir := path.Join(dir, "parrot")
+
+	cacheDir := path.Join(dir, cacheSubdir)
+
 	err = os.MkdirAll(cacheDir, os.ModeDir)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "Failed to create cacge directory")
 	}
-	return &FileCache{
+
+	return &FilesystemCache{
 		dir: cacheDir,
 		ttl: ttl,
 	}, nil
 }
 
-func (f *FileCache) GetTranslation(projectID int, languageCode, format string) ([]byte, error) {
-	s, err := os.Stat(f.key(projectID, languageCode, format))
-	if os.IsNotExist(err) {
-		return nil, ErrCacheMiss
+func (f *FilesystemCache) GetTranslation(ctx context.Context, projectID int, languageCode, format string) ([]byte, error) {
+	filename := f.key(projectID, languageCode, format)
+
+	s, err := os.Stat(filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, ErrCacheMiss
+		}
+
+		return nil, errors.Wrap(err, "Failed to get cached file state from OS")
 	}
+
 	if time.Since(s.ModTime()) > f.ttl {
+		if err := os.Remove(filename); err != nil {
+			return nil, errors.Wrap(err, "Failed to remove expired cache file")
+		}
```

Comment:
Why remove the expired file, when you're just going to overwrite it with the new contents from poedit immediately after?

Reply:
Eh... I get it. The other thing is a bit hacky

Reply:
I think you are right. I wll remove it.